apitest/scripts/trade-simulation.sh

```diff
@@ -0,0 +1,228 @@
+#! /bin/bash
+
+# Runs fiat <-> btc trading scenarios using the API CLI with a local regtest bitcoin node.
+#
+# A country code argument is used to create a country based face to face payment account for the simulated
+# trade, and the maker's face to face payment account's currency code is used when creating the offer.
+#
+# Prerequisites:
+#
+#  - Linux or OSX with bash, Java 10, or Java 11-12 (JDK language compatibility 10), and bitcoin-core (v0.19, v0.20).
+#
+#  - Bisq must be fully built with apitest dao setup files installed.
+#    Build command:  `./gradlew clean build :apitest:installDaoSetup`
+#
+#  - All supporting nodes must be run locally, in dev/dao/regtest mode:
+#           bitcoind, seednode, arbdaemon, alicedaemon, bobdaemon
+#
+#    These should be run using the apitest harness.  From the root project dir, run:
+#    `$ ./bisq-apitest --apiPassword=xyz --supportingApps=bitcoind,seednode,arbdaemon,alicedaemon,bobdaemon --shutdownAfterTests=false`
+#
+#  - Only regtest btc can be bought or sold with the test payment account.
+#
+# Usage:
+#
+#  This script must be run from the root of the project, e.g.:
+#
+#     `$ apitest/scripts/trade-simulation.sh -d buy -c fr -m 3.00 -a 0.125`
+#
+#  Script options:  -d <direction> -c <country-code> -m <mkt-price-margin(%)> - f <fixed-price> -a <amount(btc)>
+#
+# Examples:
+#
+#    Create a buy/eur offer to buy 0.125 btc at a mkt-price-margin of 0%, using an Italy face to face payment account:
+#
+#       `$ apitest/scripts/trade-simulation.sh -d buy -c it -m 0.00 -a 0.125`
+#
+#    Create a sell/eur offer to sell 0.125 btc at a fixed-price of 38,000 euros, using a France face to face
+#    payment account:
+#
+#       `$ apitest/scripts/trade-simulation.sh -d sell -c fr -f 38000 -a 0.125`
+
+export APP_BASE_NAME=$(basename "$0")
+export APP_HOME=$(pwd -P)
+export APITEST_SCRIPTS_HOME="${APP_HOME}/apitest/scripts"
+
+# Source the env and some helper functions.
+. "${APITEST_SCRIPTS_HOME}/trade-simulation-env.sh"
+. "${APITEST_SCRIPTS_HOME}/trade-simulation-utils.sh"
```

Comment:
The longform `source` might be preferable to the shortform `.` due to readability.

Reply:
Fixed in commit 0c06abdc5.